### PR TITLE
Playwright: add basic page flow spec

### LINKED ISSUE #55948: Playwright - Editor Transition: Page Basic Flow
### Details

See overall transition details here: pciE2j-sq-p2

Coverage:
- Create new page
- Accept a starting template (**this is new)
- Publish and validate

Includes coverage from prior specs:
- wp-calypso-gutenberg-page-editor-spec.js

### Checklist

_No response_

### Related

_No response_

### PR DESCRIPTION
**Pull request by dpasque**
#### Changes proposed in this Pull Request

As part of the WPCOM Playwright transition (pciE2j-sq-p2), this creates the basic page spec outlined in issue #55948.

#### Testing instructions

- [x] Gutenberg mobile passes
- [x] Gutenberg desktop passes

Closes #55948

### CODE REVIEW

File: packages/calypso-e2e/src/lib/components/page-template-modal-component.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	templateCategoryButton: ( category: TemplateCategory ) =>
+		`.page-pattern-modal__category-list button:has-text("${ category }")`,
+	mobileTemplateCategorySelectBox: '.page-pattern-modal__mobile-category-dropdown',
+	template: ( label: string ) => `button.pattern-selector-item__label:has-text("${ label }")`,
+};
+
+/**
+ * Represents the page template selection modal when first loading a new page in the editor.
+ */
+export class PageTemplateModalComponent {
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/components/page-template-modal-component.ts:17]: Do you mind moving `test/e2e/specs/specs-playwright/shared-specs/privacy-testing.ts` to use this new component as well?

File: packages/calypso-e2e/src/lib/components/editor-settings-sidebar-component.ts [hunk 2]
```diff
@@ -36,6 +36,9 @@ const selectors = {
 	addedTag: ( tagName: string ) =>
 		`${ sidebarParentSelector } .components-form-token-field:has-text("Add New Tag") .components-form-token-field__token:has-text("${ tagName }")`,
 	closeSidebarButton: `${ sidebarParentSelector } [aria-label="Close settings"]:visible`, // there's a hidden copy in there
+
+	// URl Slug
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/components/editor-settings-sidebar-component.ts]: nit: `URL`

File: packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts [hunk 3]
```diff
@@ -47,7 +47,7 @@ const selectors = {
 	// corner. This addresses the bug where the post-publish panel is immediately
 	// closed when publishing with certain blocks on the editor canvas.
 	// See https://github.com/Automattic/wp-calypso/issues/54421.
-	viewButton: 'text=/View (Post|Page)/',
+	viewButton: 'a:text-matches("View (Post|Page)")',
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts]: I think this selector is being updated in another one of my PRs too: https://github.com/Automattic/wp-calypso/pull/60864/files#diff-8aeb2361aa173b5616012d0e178c97eee06325bc9dd1b7a9570037ada0238659

File: packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts [hunk 4]
```diff
@@ -408,7 +408,7 @@ export class GutenbergEditorPage {
 	 * Publishes the post or page.
 	 *
 	 * @param {boolean} visit Whether to then visit the page.
-	 * @returns {Promise<void} No return value.
+	 * @returns {Promise<string>} The published URL.
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts:388]: Nice catch, thanks for updating 😄

File: test/e2e/specs/specs-wpcom/wp-editor__page-basic-flow.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+		: 'simpleSitePersonalPlanUser';
+
+	beforeAll( async () => {
+		page = await browser.newPage();
+
+		const testAccount = new TestAccount( accountName );
+		await testAccount.authenticate( page );
+	} );
+
+	it( 'Visit /pages page', async function () {
```
Comment (worldomonation) [inline test/e2e/specs/specs-wpcom/wp-editor__page-basic-flow.ts]: nit: I think the title should be `Visit Pages page`, by convention.

**Review by worldomonation — APPROVED** [review]
Looking great to my eyes - a few nits and a couple of suggestions, please feel free to merge after addressing them!

File: packages/calypso-e2e/src/lib/components/page-template-modal-component.ts (hunk 1 above)
Comment (dpasque) [reply packages/calypso-e2e/src/lib/components/page-template-modal-component.ts:17]: Ack, sorry I missed that! I moved it over, which means we could take the handling out of the editor page which should help with the goal of cleaning that up! Nice catch :)

File: packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts (hunk 3 above)
Comment (dpasque) [reply packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts]: Nice! I made mine match so there won't be any merge conflicts either way!

**Pull request merged**